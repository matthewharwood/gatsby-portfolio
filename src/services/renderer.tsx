function componentMapFactory() {
  const componentMap = {
    "postHeader": (props: any) => console.log("POSTHEADER", props),
    "summary": (props: any) => console.log("SUMMARY", props),
  }

  let defaultValueHandler = {
    get: function(target: any, name: any) {
      return target.hasOwnProperty(name)
        ? target[name]
        : (props: any) => console.warn(
          "We are missing a component type: ", props._type, "is not found"
        );
    },
  }

  return new Proxy(componentMap, defaultValueHandler)
}

export const pageRenderer = (raw) => {

  const componentMap = componentMapFactory();

  raw.map((d) => [d._type, d]).map(([key, props]) => componentMap[key](props));
  return true
}