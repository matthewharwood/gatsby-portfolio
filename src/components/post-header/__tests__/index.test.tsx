import React from 'react';
import renderer from 'react-test-renderer';
import { PostHeader } from '../';

const FIXTURE = {
  _key: '99cf0fa54eb7',
  _type: 'postHeader',
  eyebrow: {
    _type: 'label',
    text: 'Web Application',
  },
  mainText: {
    _type: 'label',
    text: 'Nike CMS',
  },
  projectOverview: {
    _type: 'projectOverviewSchema',
    client: {
      data: 'Google',
      type: 'default',
    },
    collaborators: {
      data: {
        items: [
          'Jimmy Soat (Creative Director)',
          'Stanley Yeung (Visual Designer)',
          'Jeffrey Qua (Developer)',
        ],
        type: 'list',
      },
    },
    company: {
      data: 'AKQA',
      type: 'default',
    },
    role: {
      data: 'Frontend Engineer',
      type: 'default',
    },
    resources: {
      data: {
        items: [
          {
            _key: '77fb5c027206',
            _type: 'link',
            href: 'https://www.github.com',
            text: 'Github',
          },
        ],
        type: 'link',
      },
    },
    technology: {
      data: {
        items: ['Angular'],
        type: 'list',
      },
    },
    when: {
      data: 'Feb 2014',
      type: 'default',
    },
  },
  subText: {
    _type: 'label',
    text: 'Some sub text here',
  },
};

describe('Page Renderer ', () => {
  it('should render postHeader', () => {
    const tree = renderer.create(<PostHeader {...FIXTURE} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
