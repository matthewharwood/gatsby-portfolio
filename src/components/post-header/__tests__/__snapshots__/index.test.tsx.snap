// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Renderer  should render postHeader 1`] = `
<div
  style={
    Object {
      "opacity": 0,
    }
  }
>
  <section
    className=""
  >
    <div
      className=""
    >
      <div
        className=""
      >
        <div
          className=""
        >
          <div
            className=""
            data-baseweb="typo-label1"
          >
            Web Application
          </div>
          <div
            className=""
            data-baseweb="typo-display2"
          >
            Nike CMS
          </div>
          <p
            className=""
            data-baseweb="typo-p1"
          >
            Some sub text here
          </p>
        </div>
        <div
          className=""
        >
          <div
            className=""
          >
            <div
              className=""
              data-baseweb="typo-label2"
              style={
                Object {
                  "textTransform": "capitalize",
                }
              }
            >
              client
            </div>
            <p
              className=""
              data-baseweb="typo-p4"
            >
              Google
            </p>
          </div>
          <div
            className=""
          >
            <div
              className=""
              data-baseweb="typo-label2"
              style={
                Object {
                  "textTransform": "capitalize",
                }
              }
            >
              collaborators
            </div>
            <p
              className=""
              data-baseweb="typo-p4"
            >
              <span>
                Jimmy Soat (Creative Director)
                <br />
              </span>
              <span>
                Stanley Yeung (Visual Designer)
                <br />
              </span>
              <span>
                Jeffrey Qua (Developer)
                <br />
              </span>
            </p>
          </div>
          <div
            className=""
          >
            <div
              className=""
              data-baseweb="typo-label2"
              style={
                Object {
                  "textTransform": "capitalize",
                }
              }
            >
              company
            </div>
            <p
              className=""
              data-baseweb="typo-p4"
            >
              AKQA
            </p>
          </div>
          <div
            className=""
          >
            <div
              className=""
              data-baseweb="typo-label2"
              style={
                Object {
                  "textTransform": "capitalize",
                }
              }
            >
              role
            </div>
            <p
              className=""
              data-baseweb="typo-p4"
            >
              Frontend Engineer
            </p>
          </div>
          <div
            className=""
          >
            <div
              className=""
              data-baseweb="typo-label2"
              style={
                Object {
                  "textTransform": "capitalize",
                }
              }
            >
              resources
            </div>
            <p
              className=""
              data-baseweb="typo-p4"
            >
              <span>
                <a
                  className=""
                  data-baseweb="link"
                  href="https://www.github.com"
                >
                  Github
                </a>
                <br />
              </span>
            </p>
          </div>
          <div
            className=""
          >
            <div
              className=""
              data-baseweb="typo-label2"
              style={
                Object {
                  "textTransform": "capitalize",
                }
              }
            >
              technology
            </div>
            <p
              className=""
              data-baseweb="typo-p4"
            >
              <span>
                Angular
                <br />
              </span>
            </p>
          </div>
          <div
            className=""
          >
            <div
              className=""
              data-baseweb="typo-label2"
              style={
                Object {
                  "textTransform": "capitalize",
                }
              }
            >
              when
            </div>
            <p
              className=""
              data-baseweb="typo-p4"
            >
              Feb 2014
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
